src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -961,23 +929,18 @@ module Base = struct
                       ~sgn
                   in
                   let%bind account_creation_fee =
-                    let num_accounts_opened =
-                      let open Field.Var in
-                      add
-                        (should_pay_for_receiver :> t)
-                        (is_empty_and_writeable :> t)
-                    in
                     let%map magnitude =
-                      Amount.Checked.scale num_accounts_opened
-                        account_creation_amount
+                      Amount.Checked.if_ is_empty_and_writeable
+                        ~then_:account_creation_amount
+                        ~else_:Amount.(var_of_t zero)
                     in
                     Amount.Signed.create ~magnitude ~sgn:Sgn.Checked.neg
                   in
                   Amount.Signed.Checked.(
                     add fee_payer_amount account_creation_fee))
```

Comment:
Should this just be the `fee_payer_amount`?

Reply:
I've updated the logic slightly above to make sure we use the default token here, and added some comments that (hopefully) clarify.

Basically, this creation fee is for fee transfers and coinbases, where this account may be receiving funds and so need to pay the fee if it doesn't already exist.